src/main/kotlin/creator/buildsystem/DirectorySet.kt

```diff
@@ -20,10 +21,11 @@ data class DirectorySet(
     val testResourceDirectory: Path
 ) {
     companion object {
-        fun create(dir: Path): DirectorySet {
-            val sourceDirectory = dir.resolve("src/main/java")
+        fun create(dir: Path, languageType: LanguageType = LanguageType.JAVA): DirectorySet {
+            val sourceDirectory = dir.resolve("src/main/" + if (languageType == LanguageType.JAVA) "java" else "kotlin")
             val resourceDirectory = dir.resolve("src/main/resources")
-            val testSourceDirectory = dir.resolve("src/test/java")
+            val testSourceDirectory =
+                dir.resolve("src/test/" + if (languageType == LanguageType.JAVA) "java" else "kotlin")
```

Comment:
You should introduce a new variable instead of having using the same if expression twice.